test/tcp.c

```diff
@@ -227,3 +227,59 @@ int test_tcp(void)
 
 	return err;
 }
+
+
+static int tcp_tos(const char *addr)
+{
+	struct tcp_test *tt;
+	struct sa srv;
+	int err;
+
+	tt = mem_zalloc(sizeof(*tt), destructor);
+	if (!tt)
+		return ENOMEM;
+
+	err = sa_set_str(&srv, addr, 0);
+	TEST_ERR(err);
+
+	err = tcp_listen(&tt->ts, &srv, tcp_server_conn_handler, tt);
+	TEST_ERR(err);
+
+	err  = tcp_settos(tt->ts, 184);
+	err |= tcp_settos(tt->ts, 120);
+	TEST_ERR(err);
```

Comment:
why is TOS set on the same tcp server two times ?

is the TOS field working for TCP server or connection ?

Reply:
> why is TOS set on the same tcp server two times ?

Mistake, removed.

> is the TOS field working for TCP server or connection ?

In this test, it is set for all incoming connections on the listening (server) TCP socket. With `tcp_conn_settos` we could also set it per connection.